app/src/main/java/com/duckduckgo/app/statistics/api/StatisticsRequester.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.statistics.api
+
+import com.duckduckgo.app.statistics.store.StatisticsDataStore
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import timber.log.Timber
+
+
+interface StatisticsUpdater {
+    fun initializeAtb()
+    fun refreshRetentionAtb()
+}
+
+class StatisticsRequester(private val store: StatisticsDataStore, private val service: StatisticsService) :
+    StatisticsUpdater {
+
+    override fun initializeAtb() {
+
+        if (store.hasInstallationStatistics) {
+            Timber.v("Atb already initialized")
+            return
+        }
+
+        service.atb()
```

Comment:
We should handle the lint warning here about not doing anything with the result of `.subscribe()`.

Also, it isn't clear why this is needed `.observeOn(AndroidSchedulers.mainThread())` as we don't appear to do anything UI thread related. Doesn't it feel more natural to call through to a `data store` on an `IO` thread anyways, rather than the UI thread?

Reply:
Good point, there's no need to jump back on the main thread so I'll remove that. As for the warning, I'll suppress it. I'm not seeing it when I run `analyze` where are you seeing it?

Reply:
Android Studio is highlighting it; maybe only happening in the newest versions of it.

<img width="662" alt="screen shot 2018-02-27 at 11 43 47" src="https://user-images.githubusercontent.com/1336281/36726989-85fad8fc-1bb3-11e8-9b32-9263c0ce0b24.png">

Reply:
Definitely not seeing it. I'll update my tools and see if that makes a difference. Are you on the beta or canary channel?

Reply:
Beta.

I can confirm it isn't flagged on AS 3.0.1 stable.

If you just want to know the suppress command, it's `@SuppressLint("CheckResult")`. If it was just a beta-quirk, I'd happily not take any action, but it looks like a valid thing to flag given it's marked with `@CheckReturnValue`.

Reply:
>I have no idea why it's complaining about that second one where I do use the value!

I don't see where it is used. There is stuff happening inside the `subscribe` block, but the return value from this isn't assigned to anything.

Reply:
In `refreshRetentionAtb` the vlaue is used in the subscribe block in `store.retentionAtb = it.version`

Reply:
That's not the complaint, though. It isn't about what is going on in the subscribe block.

The complain is that `.subscribe()` itself returns a value (specifically, a `Disposable`) and we aren't doing anything with it. 

i.e., 

    val disposable = service.updateAtb(atb, retentionAtb)
            .subscribeOn(Schedulers.io())
            .subscribe({
                Timber.v("Atb refresh succeeded")
                store.retentionAtb = it.version
            }, {
                Timber.v("Atb refresh failed with error ${it.localizedMessage}")
            }) 

This makes that lint warning go away, as you are assigning the return value. (it introduces another lint warning, though, since we don't use `disposable`.

In any case, ignoring the lint warning is the way to go.